TundraTD/Assets/Scripts/Mobs/MobPortal.cs

```diff
@@ -72,31 +81,20 @@ public void OnWaveEnded()
             if (_currentMobWaveIndex >= mobWaves.Length)
                 return;
             _currentMobWave = mobWaves[_currentMobWaveIndex];
-            infoPanel.SendNewWaveMobs(_currentMobWave);
+            infoPanel.LoadMobsInPortalPanel(_currentMobWave);
         }
 
         public void SpawnNextMob()
         {
             if (_currentMobIndex >= MobsTotalCountOnWave)
                 return;
 
-            var mobObject = _allWaveMobs[_currentMobIndex];
-            var mob = Instantiate(
-                mobObject,
-                mobSpawner.position,
-                Quaternion.identity,
-                mobSpawner.transform
-            );
+            var mob = _waveMobs[_currentMobIndex];
+            mob.RespawnMobFromPool(mobPool.transform.position);
```

Comment:
This looks a bit strange too, but now I can't say why

Reply:
In past, these lines were getting mobs and then instantiating them. Now I replaced the instantiating part with respawning current mob. Current mob is defined by currentMobIndex, which is then passed as an index into currentWave